terraform/main.tf

```diff
@@ -16,42 +16,132 @@ provider "aws" {
 
 # some local
 locals {
-  bucket_suffix   = var.environment == "production" ? "" : "-${var.environment}"
-  tf_state_bucket = "gfw-terraform${local.bucket_suffix}"
-  tags            = data.terraform_remote_state.core.outputs.tags
-  name_suffix     = terraform.workspace == "default" ? "" : "-${terraform.workspace}"
-  project         = "gfw-data-api"
+  bucket_suffix         = var.environment == "production" ? "" : "-${var.environment}"
+  tf_state_bucket       = "gfw-terraform${local.bucket_suffix}"
+  tags                  = data.terraform_remote_state.core.outputs.tags
+  name_suffix           = terraform.workspace == "default" ? "" : "-${terraform.workspace}"
+  project               = "gfw-data-api"
+  aurora_instance_class = data.terraform_remote_state.core.outputs.aurora_cluster_instance_class
+  aurora_max_vcpus      = local.aurora_instance_class == "db.t3.medium" ? 2 : local.aurora_instance_class == "db.r4.large" ? 2 : local.aurora_instance_class == "db.r4.xlarge" ? 4 : local.aurora_instance_class == "db.r4.2xlarge" ? 8 : local.aurora_instance_class == "db.r4.4xlarge" ? 16 : local.aurora_instance_class == "db.r4.8xlarge" ? 32 : local.aurora_instance_class == "db.r4.16xlarge" ? 64 : local.aurora_instance_class == "db.r5.large" ? 2 : local.aurora_instance_class == "db.r5.xlarge" ? 4 : local.aurora_instance_class == "db.r5.2xlarge" ? 8 : local.aurora_instance_class == "db.r5.4xlarge" ? 16 : local.aurora_instance_class == "db.r5.8xlarge" ? 32 : local.aurora_instance_class == "db.r5.12xlarge" ? 48 : local.aurora_instance_class == "db.r5.16xlarge" ? 64 : local.aurora_instance_class == "db.r5.24xlarge" ? 96 : ""
```

Comment:
is there no easier way in terraform to write out this logic haha?

Reply:
Ha, I wish!
I haven't found a way to expose/ extract the CPU count of the RDS instance type. If you have a better solution, please let me know